### ISSUE TITLE
False Suggestion for `const fn`

### ISSUE THREAD

**Issue by Demonstrandum** [rust-lang/rust-clippy#5192]
```sh
cargo clippy -V
#=> clippy 0.0.212 (07f1736 2020-02-11)
```
```warn
warning: this could be a `const fn`
  --> src/vm/unmarshal.rs:23:1
   |
23 | / fn fix_slice_size<T, const N : usize>(slice : &[T]) -> &[T; N] {
24 | |     let ptr = slice.as_ptr() as *const [T; N];
25 | |     unsafe { &*ptr }
26 | | }
   | |_^
   |
```
This is a false suggestion, since const parameters are not permitted in `const fn`.

**Comment by flip1995** [comment]
cc @phansch

**Comment by phansch** [comment]
This seems easy to fix, if someone wants to take it. We should be able to handle this by copying the conditional from here:

https://github.com/rust-lang/rust/blob/183e893aaae581bd0ab499ba56b6c5e118557dc7/src/librustc_ast_passes/ast_validation.rs#L570-L573

```
        if let Const::Yes(const_span) = sig.header.constness {
            // Look for const generics and error if we find any.
            for param in &generics.params {
                if let GenericParamKind::Const { .. } = param.kind {
```

and return early instead of linting here:

https://github.com/rust-lang/rust-clippy/blob/06f0ab03ef63869eb159cf7d57d5d5e826ff10ca/clippy_lints/src/missing_const_for_fn.rs#L93-L95

**Comment by krishna-veerareddy** [comment]
I can work on this one.
